### PR TITLE
docs(proposal): Falco Rules and Plugin distribution

### PR DESCRIPTION
**Pull request by leogr**
Co-authored-by: Aldo Lacuku <aldo@lacuku.eu>
Co-authored-by: Lorenzo Susini <susinilorenzo1@gmail.com>
Co-authored-by: Luca Guerra <luca@guerra.sh>
Signed-off-by: Leonardo Grasso <me@leonardograsso.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

See #136 

**Special notes for your reviewer**:

### CODE REVIEW

File: proposals/20220916-rules-and-plugin-distribution.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+Notes:
+ - Artifact's releases and versions metadata are stored in the related artifact repository.
+ - References to artifacts are similar to references to a container image (see the section below).
+
+### Reference to an artifact
+
+It can be one of:
+- name (es. `k8saudit`) -> need to use the index for lookup
```
Comment (LucaGuerra) [inline proposals/20220916-rules-and-plugin-distribution.md:82]: Should we allow to specify a tag with this or does this automatically imply ":latest" like the one below?

File: proposals/20220916-rules-and-plugin-distribution.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+For **rules files**, the problem is similar, since it's a matter of copying a specific file into a location accessible by Falco before the main executable is started.
+
+Both rules and plugins are potentially interconnected since some ruleset require specific plugins (or a class of plugins) to exist.
+
+## Goals (in scope)
+
+We want to help DevOps/DevSecOps/Security engineers who use Falco performing the following operations:
+
+* **Search** interesting plugins and rulesets for their use case, such as monitoring cloudwatch, kubernetes logs for their managed system (EKS, GCP, ...)
+* **Specify a list of plugins and rules** (ie., an index) that need to be installed from _repositories_ (which could be managed by the Falco Organization, provided by third parties or custom, both public or private) in such a way that they could be installed locally or added to Helm configuration for easy automated deployment of a Falco installation that contain the requested plugins and rules. Additionally, this functionality could enable _subscribing_ to rule repositories that may change over time and could be updated and synced without needing to redeploy Falco.
```
Comment (LucaGuerra) [inline proposals/20220916-rules-and-plugin-distribution.md:23]: `(ie., an index)` is a bit misleading I think. This requirement is to allow the user to specify what it wants to install (which can of course use the index as correctly described below)

**Comment by leogr** [comment]
cc @falcosecurity/falcoctl-maintainers

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**